arrows/mvg/camera_options.h

```diff
@@ -0,0 +1,138 @@
+
+/**
+ * The intended use of this class is for a PIMPL for an algorithm to
+ * inherit from this class to share these options with that algorithm
+ */
+struct KWIVER_ALGO_MVG_EXPORT camera_options
+{
+  /// constructor
+  camera_options();
+
+  /// copy constructor
+  camera_options( const camera_options& other );
+
+  virtual ~camera_options(){};
+
+  /// populate the config block with options
+  virtual void get_configuration( vital::config_block_sptr config ) const;
+
+  /// set the member variables from the config block
+  virtual void set_configuration( vital::config_block_sptr config );
+
+  /// option to optimize the focal length
+  bool optimize_focal_length;
+  /// option to optimize aspect ratio
+  bool optimize_aspect_ratio;
+  /// option to optimize principal point
+  bool optimize_principal_point;
+  /// option to optimize skew
+  bool optimize_skew;
+  /// the lens distortion model to use
+  LensDistortionType lens_distortion_type;
+  /// option to optimize radial distortion parameter k1
+  bool optimize_dist_k1;
+  /// option to optimize radial distortion parameter k2
+  bool optimize_dist_k2;
+  /// option to optimize radial distortion parameter k3
+  bool optimize_dist_k3;
+  /// option to optimize tangential distortions parameters p1, p2
+  bool optimize_dist_p1_p2;
+  /// option to optimize radial distortion parameters k4, k5, k6
+  bool optimize_dist_k4_k5_k6;
+  /// a soft lower bound on the horizontal field of view
+  double minimum_hfov;
+};
+
+} // namespace mvg
+
+} // namespace arrows
+
+} // namespace kwiver
+
+
+#define MVG_ENUM_HELPERS(NS, ceres_type)                                \
+namespace kwiver {                                                      \
+namespace vital {                                                       \
+                                                                        \
+template<>                                                              \
+config_block_value_t                                                    \
+config_block_set_value_cast(NS::ceres_type const& value);               \
+                                                                        \
+template<>                                                              \
+NS::ceres_type                                                          \
+config_block_get_value_cast(config_block_value_t const& value);         \
+                                                                        \
+}                                                                       \
+                                                                        \
+namespace arrows {                                                      \
+namespace mvg {                                                         \
+                                                                        \
+template<>                                                              \
+std::string                                                             \
```

Comment:
Likewise here.  Can this just be defined in the cxx?

Reply:
see declaration of MVG_ENUM_HELPERS